### PR TITLE
Add packcheck-remote to work with remote repositories directly

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: packcheck-remote.sh [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    mkdir -p "$dir" || exit 1
+
+    run_verbose "git clone $remote $dir" || exit 1
+
+    cd "$dir" || exit 1
+    run_verbose "git branch $rev" || exit 1
+
+    if test -n "$merge"
+    then
+        # This will fail is there are any merge conflicts
```
Comment (harendra-kumar) [inline packcheck-remote.sh:101]: typo "is => if"

File: packcheck-remote.sh [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        run_verbose "git merge -X theirs $merge --no-edit --commit" || exit 1
+    fi
+
+    show_step "Running packcheck"
+    # Run packcheck here
+    run_verbose "$PACKCHECK_EXE $packcheck_cli_opts"
+
+    # Go back to where the script was executed from
+    cd "$PWD" || exit 1
```
Comment (harendra-kumar) [inline packcheck-remote.sh:110]: This is not needed.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Run shellcheck on the script and fix any warnings/errors.

File: README.md [hunk 3]
```diff
@@ -199,8 +199,9 @@ $ ./packcheck-safe.sh cabal-v2 PATH=/bin:/usr/bin:/opt/ghc/bin
 
 ## packcheck-remote
 
-`packcheck-remote.sh` is a wrapper over `packcheck.sh`. It makes it easer to
-work with remote repositories.
+`packcheck-remote.sh` is a wrapper over `packcheck.sh`. It allows you to run
+packcheck on a remote repository by cloning it locally and optionally merging a
+branch into another branch (e.g. a merging a PR branch into master).
```
Comment (harendra-kumar) [inline README.md:204]: > branch into another branch (e.g. a merging a PR branch into master).

(e.g. merging a PR branch into master)

**Comment by adithyaov** [comment]
See https://github.com/composewell/packcheck/issues/37 and https://github.com/composewell/packcheck/issues/38

File: packcheck-safe.sh [hunk 4]
```diff
@@ -1,9 +1,28 @@
...
 
-PACKCHECK_DIR=$(dirname $0)
+PACKCHECK_DIR=$(dirname "$0")
 echo "Running ${PACKCHECK_DIR}/packcheck.sh with clean environment and CHECK_ENV on..."
 echo "No environment variables are honored, you have to specifiy ALL the"
 echo "parameters explicitly on the command line, including PATH."
 echo
 
-/usr/bin/env -i CHECK_ENV=y $(dirname $0)/packcheck.sh $*
+PACKCHECK_CLI_OPTS_ARR=("$@")
```
Comment (harendra-kumar) [inline packcheck-safe.sh:9]: Just quoting `$*` should work here e.g. `/usr/bin/env -i CHECK_ENV=y $(dirname $0)/packcheck.sh "$*"`

File: packcheck-remote.sh [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+            shift
+            ;;
+        *)
+            PACKCHECK_CLI_OPTS="$PACKCHECK_CLI_OPTS $i"
+            shift
+            ;;
+    esac
+done
+
+try_git_clone_and_merge "$REMOTE" "$CHECKOUT" "$MERGE" "$DIRECTORY" "$FORCE" "$PACKCHECK_CLI_OPTS"
```
Comment (harendra-kumar) [inline packcheck-remote.sh:208]: Can't we just pass `"$*"` instead of processing `$@`?

File: packcheck-safe.sh (hunk 4 above)
Comment (adithyaov) [reply packcheck-safe.sh:9]: This does not work. `a="b c d"` would become `a=b c d`.

File: packcheck-remote.sh (hunk 5 above)
Comment (adithyaov) [reply packcheck-remote.sh:208]: This does not work. `a="b c d"` would become `a=b c d`.

**Pull request merged**